mautic/mautic
# Add support for include/exclude operator for text field with clipboard support

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features" for all features
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | https://github.com/mautic/mautic-documentation/pull/236
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 




#### Description:
This PR add support for include/exclude  operator for text field in segment filters.
You are able then filter for example `city include Paris|London|New York`
Values has to be divided by separator `|`
Also it support copy from clipboard from excel column. You can copy values and paste to input. Values will be processed to `|` list of values.



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create 3 contacts with zip code: 1,2,3
3. Create segment with zip code filter include 1|2
4. Rebuild segment
5. See first two contacts are assign to segment
8. Create excel and fill 2 and 3 to column
9. Copy it
10. Go back to segment filter and paste value to zip code filter
11. You should see 2|3 
12. Reload segments
13. You should see contact 2 and contact 3 in segment

This PR require `mautic:assets:generate`




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10256"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
